form/app/Http/Controllers/ProductController.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\Product;
+class ProductController extends Controller
+{
+
+
+    //AddProduct
+   public function AddProduct(Request $req){
+        $details=new Product();
+        $details->name= $req->name;
+        $details->category= $req->category;
+        $details->price= $req->price;
+        $details->quantity= $req->quantity;
+        $req->validate([
+            'name' => 'required|string|max:255',
+            'category' =>'required ',
+            'price' => 'required|numeric|min:1',
+            'quantity'=>'required|numeric|min:1'
+
+        ]);
+        $details->save();
+        return redirect("display");
+     }
+
+
+     //DeleteProduct
+    public function DeleteProduct(Request $req){
+        $data=Product::find($req->id);
+        $data->delete();
+        return redirect("display");
+     } 
+
+
+     //UpdateProduct
+    public function UpdateProduct(Request $req){
```

Comment:
Functions should start with small letter.. like `updateProduct()`

Reply:
@Naveen1895 okey